### ISSUE TITLE
How to install redland with renv in a docker build

### ISSUE THREAD

**Issue by eblondel** [ropensci/redland-bindings#109]
Hello, in the past I've been able to install successfully _redland_ (and _rdflib_ ) in the past paying attention to install well the required libraries. Now i'm facing again the following issue of installation:

![image](https://github.com/user-attachments/assets/3d9ef2ec-5e76-4f49-a1a4-f9b80a49de9d)

I'm back on this issue, because i've moved to _renv_ to install R dependencies. With _renv_ the _redland_ installation fails.

The Docker file is available at https://github.com/fdiwg/dcf-shiny/blob/main/Dockerfile
Last tentative build: https://github.com/fdiwg/dcf-shiny/actions/runs/12257340245/job/34194580771

Your help would be much appreciated

Thanks
Emmanuel

**Comment by mbjones** [comment]
I see your Dockerfile installs `librdf0` (twice, on line 25 and line 66) and `librdf0-dev`, but can you confirm that they installed properly? I have found the following message is generally pretty accurate, and indicates that one of the underlying C libraries can't be found, or the headers can't be found:

```
#23 554.7 If redland is already installed, check that 'pkg-config' is in your
#23 554.7 PATH and PKG_CONFIG_PATH contains a redland.pc file. If pkg-config
#23 554.7 is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:
#23 554.7 R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
```

Can you find the `redland.pc` file on your image, and is it in the `PKG_CONFIG_PATH` directory?

Can you manually run `pkg-config` on the image, which should produce something like this:

```
❯ pkg-config --list-all |grep redland
redland                                         Redland RDF API Library - Redland RDF API and triple store library
```

**Comment by mbjones** [comment]
**TL;DR**: Removing `libcurl4-openssl-dev` from your Dockerfile `apt` install list made it work for me.

Details...

I tried to build your Dockerfile, and saw the same errors on my local linux docker as your GHA log. So I commented out the renv restore line, allowing the image to build (albeit incompletely). I then launched the image and exec'ed in, and can now confirm that the package containing config files for redland does not seem to be installed (`librdf0-dev`):

```sh
root@11c20892993a:/srv/dcf-shiny# pkg-config --list-all |grep redland
root@11c20892993a:/srv/dcf-shiny#
```

I then tried installing it, which tells me it needs to remove `libcurl4-openssl-dev` because they conflict. 

```
The following packages will be REMOVED:
  libcurl4-openssl-dev
```

So, my conclusion is that your `apt` install commands install `libcurl4-openssl-dev` after `librdf0-dev`, and because your Dockerfile uses `apt install -y`, it automatically removes the conflicting `librdf0-dev` package. So the solution would be to not install `libcurl4-openssl-dev` on line 63 of your Dockerfile if you don't need it. Once I manually installed `librdf0-dev`, I was able to also manually install redland in R with `install.packages("redland")` without error. 

So, I did I minor change to your Dockerfile:

```patch
diff --git a/Dockerfile b/Dockerfile
index f6d9190..61c9f3b 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -60,10 +60,8 @@ RUN apt-get update && apt-get install -y \

 RUN install2.r --error --skipinstalled --ncpus -1 redland
 RUN apt-get install -y \
-    libcurl4-openssl-dev \
     libgit2-dev \
     libxslt-dev \
-    librdf0 \
     redland-utils \
     rasqal-utils \
     raptor2-utils
```

and then rebuilt the image and the build worked for me that time and redland installed via renv without error.

I don't think there is an issue here, it was just a missing dependency.

**Comment by mbjones** [comment]
Similar problem here with that curl lib: https://forum.posit.co/t/failed-to-install-redland-package-with-github-r-lib-actions/117635/6

**Comment by eblondel** [comment]
Many thanks @mbjones much appreciated! I've changed from _libcurl4-openssl-dev_ to _libcurl4_ .
